src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,50 @@
 
 #include "t07_ip.h"
 #include <iostream>
-
+#include <string>
 
 using namespace std;
 
 int t07_ip() {
-
+    string s;
+    cin >> s;
+    int dot1=0,dot2=0,dot3=0,dot4=0;
+    bool k1=false,k2=false,k3=false,k4=false;
+    dot1=s.find('.');
+    dot2=s.find('.',dot1+1);
+    dot3=s.find('.',dot2+1);
+    dot4=s.find('.',dot3+1);
+    if (dot3!=-1 && dot4==-1 && s.find('-',0)==-1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Inefficient usage of string::find() in condition; string::starts_with() could be faster.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4487945)